lib/maven/tools/model.rb

```diff
@@ -1,377 +1,292 @@
-require 'virtus'
+require 'dry-types'
+require 'dry-struct'
+require 'dry/struct/with_setters'
 
 # keep things in line with java collections
 class Array
   def remove( *args )
     delete( *args )
   end
 end
-    
-module Maven
-  module Tools
-    Base = Virtus.model
-
-    module GAV
-      def self.included( base )
-        base.attribute :group_id, String
-        base.attribute :artifact_id, String
-        base.attribute :version, String
-      end
-    end
-    module GA
-      def self.included( base )
-        base.attribute :group_id, String
-        base.attribute :artifact_id, String
-      end
-    end
-    module SU
-      def self.included( base )
-        base.attribute :system, String
-        base.attribute :url, String
-      end
-    end
-    module NU
-      def self.included( base )
-        base.attribute :name, String
-        base.attribute :url, String
-      end
-    end
-    module INU
-      def self.included( base )
-        base.attribute :id, String
-        base.attribute :name, String
-        base.attribute :url, String
-      end
-    end
-
-    class Parent
-      include Base
-
-      include GAV
-
-      attribute :relative_path, String
-    end
-    class Organization
-      include Base
-
-      include NU
-    end
-    class License
-      include Base
-
-      include NU
-      attribute :distribution, String
-      attribute :comments, String
-    end
-    class Developer
-      include Base
 
-      include INU
-      attribute :email, String
-      attribute :organization, String
-      attribute :organization_url, String
-      attribute :roles, String
-      attribute :timezone, String
-      attribute :properties, Hash
-    end
-    class Contributor
-      include Base
-
-      include NU
-      attribute :email, String
-      attribute :organization, String
-      attribute :organization_url, String
-      attribute :roles, String
-      attribute :timezone, String
-      attribute :properties, Hash
-    end
-    class MailingList
-      include Base
+module Types
+  include Dry.Types()
+end
 
-      attribute :name, String
-      attribute :subscribe, String
-      attribute :unsubscribe, String
-      attribute :post, String
-      attribute :archive, String
-      attribute :other_archives, Array[ String ]
+module Maven
+  module Tools
+    class Base < Dry::Struct::WithSetters
```

Comment:
The library dry-types assumes immutable objects where all fields are set at construction unlike virtus. This library adds setters.